src/fetch-from-api.ts

```diff
@@ -18,9 +22,21 @@ export const fetchWithBaseUrl = async <T>(
   url: string,
   options: RequestInit,
 ): Promise<T> => {
+  const token = await getToken();
+
   const requestUrl = getUrl(url);
-  const res = await fetch(requestUrl, options);
-  const data = await getBody(res);
 
-  return { status: res.status, data } as T;
+  const customOptions: RequestInit = {
+    ...options,
+    headers: new Headers({
```

Comment:
Note that we are overwriting the other headers if there are any, sould be better to include options.headers in the new headers.